### PR TITLE
nHentai.com switch to kotlinx and add support for multilang

### LINKED ISSUE #5595: [Bug] <nHentai.com(unoriginal)>: <http error  manga>
Screenshot_20210130-193316_Tachiyomi
Extension name: nHentai.com (unoriginal)
Problem: All manga shows "http error"
Example: this manga (https://nhentai.com/zh/comic/nise-my-exciting-high-school-debut-lets-join-the-manga-club) will be redirected to this (https://nhentai.com/en/webtoon/nise-my-exciting-high-school-debut-lets-join-the-manga-club?nsfw=false) in the app. The page doesnt exist.

### PR DESCRIPTION
**Pull request by jopejoe1**
Changelog:

- switched to kotlinx
- Added support for other languages
- updated filter options
- Fixed url

Closes #5595

### CODE REVIEW

**Comment by jobobby04** [comment]
Can we keep the unoriginal naming? Just to let people know that this is a copy-cat

**Comment by jopejoe1** [comment]
> Can we keep the unoriginal naming? Just to let people know that this is a copy-cat

Yeah

**Comment by alessandrojean** [comment]
Is there any content in the website in Portuguese? I didn't managed to find it to check if it's indeed `pt` or if its `pt-BR`.

**Comment by jopejoe1** [comment]
Seems like they removed a lot of languages since the last time i checked.

**Comment by alessandrojean** [comment]
Probably it's better to only let the active languages on the extension then.

File: src/all/nhentaicom/src/eu/kanade/tachiyomi/extension/all/nhentaicom/NHentaiCom.kt [hunk 1]
```diff
@@ -113,32 +138,26 @@ class NHentaiCom : HttpSource() {
...
-            thumbnail_url = jsonObject["image_url"].nullString
-            genre = try { jsonObject["tags"].asJsonArray.joinToString { it["name"].string } } catch (_: Exception) { null }
-            artist = try { jsonObject["artists"].asJsonArray.joinToString { it["name"].string } } catch (_: Exception) { null }
-            author = try { jsonObject["authors"].asJsonArray.joinToString { it["name"].string } } catch (_: Exception) { null }
+            description = jsonObject["description"]!!.jsonPrimitive.content
+            status = SManga.COMPLETED
+            thumbnail_url = jsonObject["image_url"]!!.jsonPrimitive.content
+            genre = try { jsonObject["tags"]!!.jsonArray.joinToString { it.jsonObject["name"]!!.jsonPrimitive.content } } catch (_: Exception) { null }
+            artist = try { jsonObject["artists"]!!.jsonArray.joinToString { it.jsonObject["name"]!!.jsonPrimitive.content } } catch (_: Exception) { null }
+            author = try { jsonObject["authors"]!!.jsonArray.joinToString { it.jsonObject["name"]!!.jsonPrimitive.content } } catch (_: Exception) { null }
```
Comment (CarlosEsco) [inline src/all/nhentaicom/src/eu/kanade/tachiyomi/extension/all/nhentaicom/NHentaiCom.kt:157]: would be more kotlin to do
runCatching{jsonObject["authors"]!!.jsonArray.joinToString { it.jsonObject["name"]!!.jsonPrimitive.content}.getOrNull()

Comment (jopejoe1) [reply src/all/nhentaicom/src/eu/kanade/tachiyomi/extension/all/nhentaicom/NHentaiCom.kt]: Done

**Pull request merged**